go/vt/sqlparser/sql.y

```diff
@@ -647,19 +696,27 @@ common_table_expression:
 union_lhs:
   base_select
   {
+    if $1.HasIntoDefined() {
+      yylex.Error(fmt.Errorf("INTO clause is not allowed").Error())
+      return 1
+    }
     $$ = $1
   }
-| openb select_statement closeb
+| openb select_statement_with_no_into closeb
   {
     $$ = &ParenSelect{Select: $2}
   }
 
 union_rhs:
   base_select_no_cte
   {
+    if $1.HasIntoDefined() {
+      yylex.Error(fmt.Errorf("INTO clause is not allowed").Error())
+      return 1
+    }
     $$ = $1
   }
-| openb select_statement closeb
+| openb select_statement_with_no_into closeb
```

Comment:
(minor) You might consider `select_statement_with_no_trailing_into`. It's a little more verbose, but it captures some of the complexity that into can appear in multiple places in a select statement and that rule is preventing it from occurring at the end. Not a big deal either way though, just a thought as I was re-reading the code.